config/src/main/java/com/typesafe/config/impl/SimpleConfigOrigin.java

```diff
@@ -550,4 +550,18 @@ static SimpleConfigOrigin fromBase(SimpleConfigOrigin baseOrigin,
         Map<SerializedField, Object> fields = applyFieldsDelta(baseFields, delta);
         return fromFields(fields);
     }
+
+    @Override
+    public ConfigOrigin withComments(List<String> comments) {
```

Comment:
here I think we could just rename setComments to withComments. The codebase is inconsistent about `set` vs. `with`, but I think I like `with` better.